widgetry/src/text.rs

```diff
@@ -46,26 +47,31 @@ impl Font {
 #[derive(Debug, Clone)]
 pub struct TextSpan {
     text: String,
-    fg_color: Color,
+    fg_color: Option<Color>,
     size: usize,
     font: Font,
     underlined: bool,
 }
 
 impl TextSpan {
     pub fn fg(mut self, color: Color) -> TextSpan {
-        assert_eq!(self.fg_color, DEFAULT_FG_COLOR);
-        self.fg_color = color;
+        assert_eq!(self.fg_color, None);
+        self.fg_color = Some(color);
         self
     }
+
     pub fn maybe_fg(mut self, color: Option<Color>) -> TextSpan {
-        assert_eq!(self.fg_color, DEFAULT_FG_COLOR);
+        assert_eq!(self.fg_color, None);
         if let Some(c) = color {
```

Comment:
Nit: Simpler to just do `self.fg_color = color` now

Reply:
Indeed! Fixed.